components/provisioner/internal/apperrors/apperrors.go

```diff
@@ -9,42 +9,58 @@ const (
 	CodeBadRequest ErrCode = 400
 )
 
+const (
+	IntCodeBadGateway     IntErrCode = 10
+	IntCodeInternal       IntErrCode = 11
+	IntCodeForbidden      IntErrCode = 12
+	IntCodeBadRequest     IntErrCode = 13
+	IntCodeTenantNotFound IntErrCode = 13
+)
+
 type ErrCode int
 
+type IntErrCode int
+
 type AppError interface {
```

Comment:
We could make the solution more readable by applying the following change:
```
type AppError interface {
        Append(string, ...interface{}) AppError
	Code() ErrCode
	Cause() CauseCode
	Error() string
}
```
If we use "Cause" term the meaning of the additional error code gets obvious (Internal Code is enigmatic a bit).